### ISSUE TITLE
Bad file install locations

### ISSUE THREAD

**Issue by sambler** [AcademySoftwareFoundation/OpenShadingLanguage#1771]
### Problem

The 1.13.6.1 release installs some new files that are not in acceptable locations.

All `.cmake` files should be installed into one location. Extra support scripts should be installed under `share` not at the top of the filesystem.

**Expected behavior:**

A clean install with files in expected locations

**Actual behavior:**

cmake config files and extra support scripts are installed into two locations.

### Versions

* OSL branch/version: 1.13.6.1

### Solution

`llvm_macros.cmake` should have `DESTINATION ${OSL_CONFIG_INSTALL_DIR}` to put it with the other cmake files.

The `serialize-bc.py` script would be better installed somewhere like `DESTINATION ${CMAKE_INSTALL_DATADIR}/build-scripts`

**Comment by GvMariani** [comment]
Confirmed here, with latest 1.13.7.0: llvm_macros.cmake ends up in /usr/cmake and serialize-bc.py in /usr/build-scripts.

The llvm_macros.cmake place you propose is OK AFAICT, but the serialize-bc.py one does not look right: if I'm not reading wrongly, llvm_macroscmake calls serialize-bc.py with the following statement:
`# Serialize the linked bitcode into a CPP file 

    set ( src_bc_cpp "${CMAKE_CURRENT_BINARY_DIR}/${output_name}.bc.cpp" )

    add_custom_command ( OUTPUT ${src_bc_cpp}
        COMMAND ${Python_EXECUTABLE} "${_THIS_MODULE_BASE_DIR}/../build-scripts/serialize-bc.py"
            ${linked_src_bc} ${src_bc_cpp} ${output_name}
        DEPENDS "${_THIS_MODULE_BASE_DIR}/../build-scripts/serialize-bc.py" ${linked_src_bc}
        ${exec_headers} ${PROJECT_PUBLIC_HEADERS}
        WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}" )`

So you cannot put the python file everywhere: the build-scripts should go in a directory above the one where llvm_macros.cmake resides, otherwise the above command won't work...
However this is a weird path for python files, at least in my distro.

**Comment by jfpanisset** [comment]
Agreed about `llvm_macros.cmake`, when I build the [aswf-docker images](https://github.com/AcademySoftwareFoundation/aswf-docker) the file ends up in `/usr/local/cmake/llvm_macros.cmake` instead of where the other OSL CMake files land, in `/usr/local/lib/cmake/OSL/`

The following patch against 1.15.6.0 would land the file in the same location as the other OSL CMake files:

```
--- CMakeLists.txt.orig 2026-09-02 20:34:27.776680531 -0700
+++ CMakeLists.txt      2026-09-02 20:36:23.346443574 -0700
@@ -324,7 +324,7 @@
 install (FILES "${OSL_PROJECT_CONFIG}" "${OSL_VERSION_CONFIG}"
         DESTINATION "${OSL_CONFIG_INSTALL_DIR}")

-install (FILES src/cmake/llvm_macros.cmake DESTINATION cmake)
+install (FILES src/cmake/llvm_macros.cmake DESTINATION ${OSL_CONFIG_INSTALL_DIR}")

 set (PERMISSION_FLAGS OWNER_EXECUTE OWNER_READ OWNER_WRITE
                       GROUP_EXECUTE GROUP_READ
```

But more generally: this file seems to only contain the definition of a function:

```
function ( EMBED_LLVM_BITCODE_IN_CPP src_list suffix output_name list_to_append_cpp extra_clang_args include_dirs)
...
```

which isn't referenced by any of the CMake files installed in `lib/cmake/OSL/`, so is the file needed at all by OSL clients, and if not, can its installation be skipped?

**Comment by lgritz** [comment]
I think that for sure, this should end up in the cmake/OSL area with the rest of the installed cmake components, if we install it at all.

I'm not sure about the use of the function you mention. It's used within the project, of course. I'm not sure if it's used by downstream clients. It is used in testshade/CMakeLists.txt, so that makes me think it's at least possibly meant for downstream consumption. OTOH, testrender/CMakeLists.txt does not seem to use it, so I'm not sure.

Let's ask on the slack or mail list and see if anybody says they use it? If not, we can try not installing it and see if anybody complains. Easy enough to bring back.
